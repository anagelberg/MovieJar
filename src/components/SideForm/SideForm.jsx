import './SideForm.scss';
import ClosingX from '../ClosingX/ClosingX';

function SideForm({ onClose, form }) {

    return (
        <div className='side-form'>
            <ClosingX closeHandler={onClose} modifier='--light' />
            {form()}
        </div>
    )

}

export default SideForm;