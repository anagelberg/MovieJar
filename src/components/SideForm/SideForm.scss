@use "../../styles/variables" as *;
@use "../../styles/mixins" as *;

.side-form {
  height: calc(100vh - $nav-height-mob);
  width: 100vw;
  overflow-y: auto;
  background-color: $secondary-background;
  box-shadow: $box-shadow;
  position: fixed;
  left: 0;
  bottom: 0;
  display: flex;
  flex-direction: column;
  padding: 0.5rem;
  z-index: 10;

  @include tablet {
    width: calc($sidebar-width + $spacing-unit-lg); //fix this too
    height: calc(100vh - $nav-height-tab);
  }

  &__close {
    @include small-icon;
    align-self: flex-end;
  }
}
