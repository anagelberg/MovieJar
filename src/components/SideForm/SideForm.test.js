import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import SideForm from './SideForm';

describe('SideForm Component Tests', () => {

    const mockOnClose = jest.fn();
    const mockFormContent = 'Form Content';
    const mockForm = () => mockFormContent;

    it('renders the form content without crashing', () => {
        render(<SideForm onClose={mockOnClose} form={mockForm} />);
        expect(screen.getByText(mockFormContent)).toBeInTheDocument();
    });

    it('matches the snapshot', () => {
        const { asFragment } = render(<SideForm onClose={mockOnClose} form={mockForm} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('calls onClose prop when ClosingX is clicked', () => {
        render(<SideForm onClose={mockOnClose} form={mockForm} />);
        const closeX = screen.getByTestId('closing-x');
        fireEvent.click(closeX);
        expect(mockOnClose).toHaveBeenCalled();
    });



});
