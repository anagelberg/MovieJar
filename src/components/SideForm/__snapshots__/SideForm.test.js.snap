// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideForm Component Tests matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="side-form"
  >
    <svg
      class="close-x close-x--light"
      data-testid="closing-x"
    >
      close.svg
    </svg>
    Form Content
  </div>
</DocumentFragment>
`;
